instance/tests/models/test_instance.py

```diff
@@ -335,6 +340,29 @@ def test_run_playbook(self, mock_open_repo, mock_run_playbook, mock_inventory, m
             username='ubuntu',
         ), mock_run_playbook.mock_calls)
 
+    @patch('instance.models.instance.ansible.run_playbook')
+    def test_run_playbook_logging(self, mock_run_playbook):
+        """
+        Ensure logging routines are working on _run_playbook method
+        """
+        with tempfile.NamedTemporaryFile() as stdout:
+            with tempfile.NamedTemporaryFile() as stderr:
+
+                mock_run_playbook.return_value.__enter__.return_value.stdout = open(stdout.name, "rb")
+                mock_run_playbook.return_value.__enter__.return_value.stderr = open(stderr.name, "rb")
+                mock_run_playbook.return_value.__enter__.return_value.returncode = 0
+
+                stdout.write(b"HELLO\n")
+                stdout.flush()
+                stderr.write(b"HI\n")
+                stderr.flush()
+
+                instance = OpenEdXInstanceFactory()
+                log, returncode = instance._run_playbook("requirements", "playbook")
+
+                self.assertListEqual(sorted(log), sorted(['HELLO', 'HI']))
```

Comment:
@chromano Djangos `TestCase` class derives from `unittest.TestCase`.  The problem is that the method has been renamed in Python 3 to `self.assertCountEqual()` – I'm still mostly using Python 2, so I forgot about this.